src/tests/ftest/osa/osa_offline_reintegration.py

```diff
@@ -30,98 +24,87 @@ def setUp(self):
-
 
     def run_offline_reintegration_test(self, num_pool, data=False,
-                                       server_boot=False):
+                                       server_boot=False, oclass=None,
+                                       reint_during_rebuild=False,
+                                       reint_during_aggregation=False):
         """Run the offline reintegration without data.
             Args:
             num_pool (int) : total pools to create for testing purposes.
             data (bool) : whether pool has no data or to create
                           some data in pool. Defaults to False.
             server_boot (bool) : Perform system stop/start on a rank.
                                  Defults to False.
+            oclass (str) : daos object class string (eg: "RP_2G8")
+            reint_during_rebuild (bool) : Perform reintegration during
+                                          rebuild (Defaults to False).
+            reint_during_aggregation (bool) : Perform reintegration
+                                              during aggregation
+                                              (Defaults to False).
         """
         # Create a pool
         pool = {}
         pool_uuid = []
-        exclude_servers = (len(self.hostlist_servers) * 2) - 1
+        if oclass is None:
+            oclass = self.ior_cmd.dfs_oclass.value
 
-        # Exclude rank : two ranks other than rank 0.
-        rank = random.randint(1, exclude_servers)
+        # Exclude ranks [0, 3, 4]
+        rank = [0, 3, 4]
 
         for val in range(0, num_pool):
             pool[val] = TestPool(self.context,
                                  dmg_command=self.get_dmg_command())
             pool[val].get_params(self)
-            # Split total SCM and NVME size for creating multiple pools.
-            pool[val].scm_size.value = int(pool[val].scm_size.value /
-                                           num_pool)
-            pool[val].nvme_size.value = int(pool[val].nvme_size.value /
-                                            num_pool)
             pool[val].create()
             pool_uuid.append(pool[val].uuid)
             self.pool = pool[val]
+            if reint_during_aggregation is True:
+                test_seq = self.ior_test_sequence[1]
+                self.pool.set_property("reclaim", "disabled")
+            else:
+                test_seq = self.ior_test_sequence[0]
             if data:
-                self.run_ior_thread("Write", self.ior_dfs_oclass[0],
-                                    self.ior_apis[0], self.ior_test_sequence[0])
+                self.run_ior_thread("Write", oclass, test_seq)
+                self.run_mdtest_thread()
 
-        # Exclude and reintegrate the pool_uuid, rank and targets
-        for val in range(0, num_pool):
-            self.pool = pool[val]
+        # Exclude all the ranks
+        random_pool = random.randint(0, (num_pool-1))
+        for val in range(0, len(rank)):
```

Comment:
(pylint-consider-using-enumerate) Consider using enumerate instead of iterating with range and len